lua/ttt2/libraries/targetid.lua

```diff
@@ -473,9 +473,9 @@ function targetid.HUDDrawTargetIDRagdolls(tData)
 
 	local corpse_found = CORPSE.GetFound(ent, false) or not DetectiveMode()
 	local corpse_ply = corpse_found and CORPSE.GetPlayer(ent) or false
-	local role_found = (corpse_found and ent.bodySearchResult and ent.bodySearchResult.subrole) or (corpse_ply and corpse_ply:GetSubRole())
 	local binoculars_useable = IsValid(c_wep) and c_wep:GetClass() == "weapon_ttt_binoculars" or false
-	local roleData = (corpse_ply and corpse_ply:GetSubRoleData()) or roles.GetByIndex(role_found and ent.bodySearchResult.subrole or ROLE_INNOCENT)
+	local role_found = (corpse_found and ent.bodySearchResult and ent.bodySearchResult.subrole) or (corpse_ply and IsValid(corpse_ply) and corpse_ply:GetSubRole())
+	local roleData = (corpse_ply and IsValid(corpse_ply) and corpse_ply:GetSubRoleData()) or roles.GetByIndex(role_found and ent.bodySearchResult.subrole or ROLE_INNOCENT)
```

Comment:
```suggestion
	local role_found = (corpse_found and ent.bodySearchResult and ent.bodySearchResult.subrole) or (IsValid(corpse_ply) and corpse_ply:GetSubRole())
	local roleData = (IsValid(corpse_ply) and corpse_ply:GetSubRoleData()) or roles.GetByIndex(role_found and ent.bodySearchResult.subrole or ROLE_INNOCENT)
```

IsValid already checks for nil afaik